docs/section19/Rule 19-39.md

```diff
@@ -0,0 +1,77 @@
+
+        - Create aggregated OA CFM schedule for the baseline for the zones served by the hvac system: `aggregated_min_OA_schedule_across_zones_b = aggregate_min_OA_schedule_across_zones(B_RMI,zone_OA_CFM_list_of_schedules_b)`  
+        - Create aggregated OA CFM schedule for the proposed for the zones served by the baseline hvac system: `aggregated_min_OA_schedule_across_zones_p = aggregate_min_OA_schedule_across_zones(P_RMI,zone_OA_CFM_list_of_schedules_p)`  
+        - Reset OA_CFM_schedules_match boolean variable to True: `OA_CFM_schedules_match = True`  
+        - Check if schedules mismatch for any hours of the year, if so set boolean to false: `for p to Range(8760):`  
+            - Check if the baseline and proposed schedules mismatch: `if aggregated_min_OA_schedule_across_zones_b[p] != aggregated_min_OA_schedule_across_zones_p[p]: OA_CFM_schedule_match = false`   
+        - Sum OA CFM across baseline schedule to get the total for comparison purposes in the rule assertions: `modeled_baseline_total_zone_min_OA_CFM = sum(aggregated_min_OA_schedule_across_zones_b)`  
+        - Sum OA CFM across proposed schedule to get the total for comparison purposes in the rule assertions: `modeled_proposed_total_zone_min_OA_CFM = sum(aggregated_min_OA_schedule_across_zones_p)`  
+
+        - Get the maximum_outdoor_airflow for the fan system: `max_OA_flow = hvac.fan_system.maximum_outdoor_airflow`  
+        - Get the design supply air flow (1st element of the list): `design_supply_flow = list(get_fan_system_object_supply_return_exhaust_relief_total_kW_CFM(B_RMI, hvac.fan_system)["Fan_CFM])[0]`  
+        - Get the minimum fan system air flow: `min_air_flow = hvac.fan_system.minimum_airflow`  
+        - Get the minimum fan system OA flow: `min_OA_flow = hvac.fan_system.minimum_outdoor_airflow` 
+        - Check if it was modeled as a 100% outdoor air unit (was not sure how to test this exactly, thought this might work although may we need a data element in the schema for this unless I missed it: `if if max_OA_flow == design_supply_flow and min_OA_flow == min_air_flow: modeled_100_percent_OA = true`  
+        
+    - **Rule Assertion:** 
+    - Case 1: If the baseline system was modeled with the design supply cfm equal to the maximum OA cfm and the minimum air flow was modeled as the minimum OA flow and all space types are defined and are all lab spaces then pass: `if modeled_100_percent_OA == true and modeled_baseline_total_zone_min_OA_CFM >= modeled_proposed_total_zone_min_OA_CFM and all_lighting_space_types_defined== true: outcome = "PASS" and raise_message "The proposed system serves lab spaces and is modeled as a 100% outdoor air system. The corresponding baseline system is also modeled as an 100% outdoor air system. This meets G3.1.2.5 Exception 4 for hvac systems that are prohibited from recirculating return air by code or accreditation standards. Recommend a manual check to confirm that recirculating return air is prohibited."`  
```

Comment:
@mkarpman Do you agree that I need to add logic to check that the proposed is also modeled as 100% OA. I don't know how I missed this originally so I wonder if I am forgetting something. This would mean that I would need to check if all zones served by the baseline HVAC system are served by a 100% OA system in the proposed.

Perhaps I just made this assumption because the proposed system serves labs?

Reply:
I think we do need to check proposed, because labs may not necessarily have 100% OA (as far as I know).